packages/langium/src/lsp/goto.ts

```diff
@@ -45,34 +45,37 @@ export class DefaultGoToResolverProvider implements GoToResolver {
 
     goToDefinition(document: LangiumDocument, params: DefinitionParams): MaybePromise<LocationLink[] | undefined> {
         const rootNode = document.parseResult.value;
-        const targetCstNodes: GoToLink[] = [];
         if (rootNode.$cstNode) {
             const cst = rootNode.$cstNode;
             const sourceCstNode = findLeafNodeAtOffset(cst, document.textDocument.offsetAt(params.position));
             if (sourceCstNode) {
-                const goToLink = this.findLink(sourceCstNode);
-                if (goToLink) {
-                    targetCstNodes.push(goToLink);
-                }
+                return this.collectLocationLinks(sourceCstNode, params);
             }
         }
-        return targetCstNodes.map(link => LocationLink.create(
-            link.targetDocument.textDocument.uri,
-            (this.findActualNodeFor(link.target) ?? link.target).range,
-            link.target.range,
-            link.source.range
-        ));
+        return undefined;
+    }
+
+    protected collectLocationLinks(sourceCstNode: LeafCstNode, _params: DefinitionParams): MaybePromise<LocationLink[] | undefined> {
```

Comment:
Suggestion: This should be a general `CstNode`.
```suggestion
    protected collectLocationLinks(sourceCstNode: CstNode, _params: DefinitionParams): MaybePromise<LocationLink[] | undefined> {
```